packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFScrollViewDelegateHostApi.m

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FWFScrollViewDelegateHostApi.h"
+#import "FWFWebViewHostApi.h"
+
+@interface FWFScrollViewDelegateFlutterApiImpl ()
+// BinaryMessenger must be weak to prevent a circular reference with the host API it
+// references.
+@property(nonatomic, weak) id<FlutterBinaryMessenger> binaryMessenger;
+// InstanceManager must be weak to prevent a circular reference with the object it stores.
+@property(nonatomic, weak) FWFInstanceManager *instanceManager;
```

Comment:
can you explain more about this circular reference?

Reply:
Sorry, but that comment is actually not come up by me, you can see that comment throughout the code base wherever the property of `FWFInstanceManager ` is declared, I just followed the convention. 

However, from my understanding, it is a safety measure to prevent a memory leak in case the instance of `FWFInstanceManager` is used to store the object that contains it (circular relation).

Reply:
@hellohuanlin This is the pattern for all the api implementations in the plugin. The full reasoning is that the `ScrollViewDelegate` instance contains an instance of the `ScrollViewDelegateFlutterApi`. And the `ScrollViewDelegateFlutterApi` contains an instance of the `FWFInstanceManager`. And the `FWFInstanceManager` contains an instance of the `ScrollViewDelegate`.

`ScrollViewDelegate` -> `FlutterApi` -> `InstanceManager` -> `ScrollViewDelegate`. 

I'm currently working on https://github.com/flutter/flutter/issues/134777 to eventually make this a part of generated code and not copy pasted.